### ISSUE TITLE
Undefined Symbol  RCTMGLtileSource

### ISSUE THREAD

**Issue by L-Yeiser** [react-native-mapbox-gl/maps#417]
**Describe the bug**
Upgrading to `7.0.5` with manual linking causes builds to fail with the error `Undefined symbol: _OBJC_CLASS_&_RCTMGLTILE  SOURCE`

![Screen Shot 2019-09-25 at 12 14 09 PM](https://user-images.githubusercontent.com/6750498/65632668-178dd800-df8f-11e9-891d-1d3d6e5be0ce.png)

The breaking change was introduced in [this merge commit](a0cf6d38c5744ebeade8de527388f6a979ff62f0). [Forking the library and reverting that commit](https://github.com/L-Yeiser/maps/tree/revert-breaking-commit) fixes the build error.

**To Reproduce**
Upgrade to `7.0.5` with manual linking. Attempt to build.

**Expected behavior**
The project should build.

**Versions (please complete the following information):**
 - Platfrom: iOs
- XCode: 11.0
 - React Native Version: 0.59

**Comment by mattijsf** [comment]
Hi. Can you try running `pod install` from the `/ios` then rebuild your app?

**Comment by kristfal** [comment]
Thanks for notifying, we should fix this or notify about this in changelogs. I use pods, which seem to work just fine. 

I do have an idea why this happens though. Can you try to unlink and relink after upgrading?

**Comment by mattijsf** [comment]
@kristfal I think this is because I didn't update `RCTMGL.xcodeproj` with the new files added in #402 I was using the example project / pods when testing and didn't see this. I'll create a new PR for it.

**Comment by kristfal** [comment]
Ok, good. Can you also increment version to 7.0.6 to prepare for an NPM release? Once it is merged, all you’ll need to do is push a new tag 7.0.6 onto master via git and that will trigger the publish. Also: Is there also a way we could catch this issue on the CI? It does sound like it’s easy to miss.

**Comment by mattijsf** [comment]
I'll wait for green lights on CI. Then I'll tag it once merged.

I was also thinking about that, some approaches come to mind
- Build separate project that builds `RCTMGL.xcodeproj` (+ React). Probably slows done CI significantly.
- Cocoapods generates a project using `react-native-mapbox-gl.podspec` / `ios/RCTMGL/**/*.{h,m}`. We could write a simple sanity check that validates if `RCTMGL.xcodeproj` references all these files too.

**Comment by mattijsf** [comment]
@L-Yeiser 7.0.6 is now available from NPM which should fix this issue

**Comment by L-Yeiser** [comment]
Thanks so much for getting to this so quickly.
